### ISSUE TITLE
Distinguish `stable` from `latest` documentation

### ISSUE THREAD

**Issue by astrojuanlu** [kedro-org/kedro#2823]
## Description
At the moment, `stable` and `latest` versions of the docs are identical visually speaking, which might confuse users. cc @stichbury 

## Context
Today a user was confused because they thought that `latest` meant "latest stable version", when in fact means "latest development version" https://www.linen.dev/s/kedro/t/13226075/dataset-factory-hi-everyone-i-m-experimenting-with-dataset-f#2842ff79-1d6c-4859-a5f9-f93fcfa50ca8

(also cc @humitos can `latest` be renamed on RTD? or is it hardcoded?)

## Possible Implementation
Add version warning for `latest` docs, like the one we have for older versions:

![image](https://github.com/kedro-org/kedro/assets/316517/0a151dec-52fa-4510-91f7-c9121dbb8ac7)

## Possible Alternatives

❓

**Comment by stichbury** [comment]
How would this work so it was only ever on `latest`? Would we be able to control this through RTD or have to add it manually and remove it again for releases, then add it back for the next?

**Comment by humitos** [comment]
> (also cc @humitos can `latest` be renamed on RTD? or is it hardcoded?)

You can take different approaches, depending on you goal:

* make `latest` to point to whatever branch/tag you want from the project's setting by defining the "Default branch"
* disable `latest` version completely if you want
* set "Default version" to something different than `latest` (this is the "Default version" setting)
* override `latest` by creating a branch/tag with that name

**Comment by noklam** [comment]
@humitos I guess it is because "latest" refers to our "main" branch while stable refer to the latest release.

Some user try to find docs in latest and mistaken it refers to the latest version.

**Comment by datajoely** [comment]
Can we rename `latest` , `nightly`? is that clearer that it's WIP?

**Comment by astrojuanlu** [comment]
> Can you expand a little more on the goal?

It looks like "latest" means "latest stable version" to some users, so we'd like to know if it's possible to rename it to something like `nightly`, `unreleased`, or similar.

(I think "nightly" is very obvious for developers but not so much for users)

**Comment by astrojuanlu** [comment]
In any case, this issue has two parts: renaming `latest` (if possible) and more importantly adding some visual element that tells the user that they're reading documentation of features that aren't yet available in a released version of Kedro (should be easy)

**Comment by astrojuanlu** [comment]
Renaming `latest` is not an option.

About the visual cues, we can either wait until RTD solves the search-as-you-type glitches of the add-ons and make use of https://github.com/readthedocs/addons/pull/125, or use https://sphinx-version-warning.readthedocs.io/ instead.

**Comment by humitos** [comment]
> About the visual cues, we can either wait until RTD solves the search-as-you-type glitches of the add-ons and make use of [readthedocs/addons#125](https://github.com/readthedocs/addons/pull/125), 

This is fixed and it should be released in the next few days.

> or use [sphinx-version-warning.readthedocs.io](https://sphinx-version-warning.readthedocs.io/) instead.

I don't think I will keep maintaining this extension anymore in the future, since it will be superseded by the addons notification.

**Comment by astrojuanlu** [comment]
Thanks to the new RTD add-ons, this is already being done.

**Comment by astrojuanlu** [comment]
Btw, now that https://github.com/readthedocs/readthedocs.org/issues/10674 was addressed, I went ahead and enabled the `main` version and renamed it to `nightly`.

This means that now we should have 2 versions: `stable` and `nightly`. Hence the `latest` version, which was very confusing, should remain hidden. cc @ankatiyar @DimedS 

About `stable`, I still think it should be a redirection:

> And on the topic of the flyout, here's my thinking:
> 
> I think I have become numb to the whole `stable/latest` from RTD, but I think @stichbury is right this is not at all obvious.
> 
> Now, look at what happens when I change the default version to be `0.19.3` instead of the current `stable`:
> 
> 1. User types `https://docs.kedro.org`
> 2. User gets redirected to `https://docs.kedro.org/en/0.19.3/`
> 3. The flyout shows the number:
> 
> ![Screenshot 2024-03-26 at 15-34-38 Welcome to Kedro’s award-winning documentation! — kedro 0 19 3 documentation](https://github.com/kedro-org/kedro/assets/316517/1d253d9c-f6bf-437d-a25c-770022015c34)
> 
> By having the number in the URL and also in the flyout _by default_, I think it's more obvious how the user should go and switch to their version of choice.

 _Originally posted by @astrojuanlu in [#3741](https://github.com/kedro-org/kedro/issues/3741#issuecomment-2020767653)_

**Comment by astrojuanlu** [comment]
Amazing ❤ https://ai.pydantic.dev/

![Image](https://github.com/user-attachments/assets/290646c3-cba0-473d-9cd7-7be9c120f485)

**Comment by astrojuanlu** [comment]
Version notice from Pydantic AI docs was removed in https://github.com/pydantic/pydantic-ai/pull/815

**Comment by astrojuanlu** [comment]
My current stance is as follows:

- [x] P0: ~~latest~~ is permanently redirected to `/unreleased` (explicit, unambiguous) (https://github.com/readthedocs/readthedocs.org/issues/10674)
- [x] P1: `stable` becomes a dynamic redirection to the most recent released version (so that it's always visible in the URL and the flyout) (https://github.com/readthedocs/readthedocs.org/issues/11183#issuecomment-2032403457)
- [ ] P2: We add some custom logic to our MkDocs docs to warn readers of old and unreleased versions

Thoughts?

**Comment by datajoely** [comment]
I like unreleased for the semantics you've laid out.

I've seen some libraries like PydanticAI only show the latest unreleased version as a signal to users they should be looking to the future, I'm not sure we need to do the same but it's an interesting idea.

**Comment by datajoely** [comment]
I also think `current` or `current release` is unambiguous like unreleased is.

**Comment by astrojuanlu** [comment]
> I've seen some libraries like PydanticAI only show the latest unreleased version as a signal to users they should be looking to the future, I'm not sure we need to do the same but it's an interesting idea.

Yeah I'm not so confident with making it more difficult for Kedro users to read old docs, given that there's lots of projects out there with old versions. I think PydanticAI can afford it because it's a nascent library that's about to release 1.0 https://github.com/pydantic/pydantic-ai/issues/1974

**Comment by DimedS** [comment]
Thanks for your proposal and clear explanation, @astrojuanlu.
Personally, I don’t see a major issue with the name "latest", because when it appears alongside "stable", it becomes quite clear that "latest" likely refers to an unreleased version.

That said, I’m happy to support renaming "latest" to "unreleased" if it has caused confusion for some of our users.
Just to confirm - "stable" will remain as it is, correct?

**Comment by astrojuanlu** [comment]
I also didn't see an issue with `/latest` but maybe because I became numb to it. It was highlighted by @stichbury and also in user questions https://www.linen.dev/s/kedro/t/13226075/dataset-factory-hi-everyone-i-m-experimenting-with-dataset-f#2842ff79-1d6c-4859-a5f9-f93fcfa50ca8

Both `/stable` and `/latest` remain (otherwise, thousands of links would break). But they become redirections.

- When the user types `/latest`, they will be redirected to `/unreleased`
- When the user types `/stable`, they will be redirected to `/0.19.14` or whatever appropriate version there is.

**Comment by DimedS** [comment]
> * When the user types `/stable`, they will be redirected to `/0.19.14` or whatever appropriate version there is.

Got it - so it won’t be a dedicated copy of the docs anymore. At the moment, `/stable` is a full duplicate of `/0.19.14`. I support this change - I think it makes it much clearer which exact version is considered stable.

We used `data.versions.current.slug == "stable"` to determine whether the current version is the stable one, so we could avoid adding noindex tags. This logic will likely need to be updated.

**Comment by Huongg** [comment]
I like this, however as far as I know, kedro-datasets docs only has the `latest` but not the `stable`. Is it going to be an issue?

**Comment by astrojuanlu** [comment]
Glad you asked 😄 The same redirection trick can be applied in `kedro-datasets`, which will _finally_ bring consistency among all subprojects.

**Comment by astrojuanlu** [comment]
`/en/latest` is now a HTTP 301 redirection to `/en/unreleased` in all 3 projects ✔

**Comment by astrojuanlu** [comment]
`/en/stable` is now a HTTP 302 dynamic redirection to the most recent released version in all 3 projects ✔
